ci-operator/step-registry/telcov10n/sno-hub-ztp/share-hub-cluster-profile/telcov10n-sno-hub-ztp-share-hub-cluster-profile-commands.sh

```diff
@@ -0,0 +1,120 @@
+    -o 'UserKnownHostsFile=/dev/null'
+    -o 'ServerAliveInterval=90'
+    -o LogLevel=ERROR
+    -i "${CLUSTER_PROFILE_DIR}/ssh-key"
+  )
+
+}
+
+function create_shared_hub_cluster_profile_artifacts_folder {
+  
+  echo "************ telcov10n Create a shared folder for those artifacts that would be used during Spoke deployments ************"
+
+  timeout -s 9 10m ssh "${SSHOPTS[@]}" "root@${AUX_HOST}" bash -s --  \
+  "${NAMESPACE}" "${SHARED_HUB_CLUSTER_PROFILE}" << 'EOF'
+set -o nounset
+set -o errexit
+set -o pipefail
+
+cluster_profile_shared_folder=/var/builds/${1}/${2}
+cluster_profile_symbolic_link=/var/telco-qe-preserved/${2}
+
+test -f ${cluster_profile_symbolic_link} && rm -fr ${cluster_profile_symbolic_link}
+mkdir -p ${cluster_profile_shared_folder}
+link -sf ${cluster_profile_shared_folder} ${cluster_profile_symbolic_link}
+EOF
+}
+
+function save_hub_cluster_profile_artifacts {
+
+  echo "************ telcov10n Save those artifacts that will be used during Spoke deployments ************"
+
+  hub_to_spoke_artifacts=(
+    "${KUBECONFIG}"
+    "${KUBEADMIN_PASSWORD_FILE}"
+    "${CLUSTER_PROFILE_DIR}"
+  )
+
+  echo
+  set -x
+  rsync -avP \
+    -e "ssh $(echo "${SSHOPTS[@]}")" \
+    "${hub_to_spoke_artifacts[@]}" \
+    "root@${AUX_HOST}":/var/telco-qe-preserved/${SHARED_HUB_CLUSTER_PROFILE}/
+  set +x
+  echo
+}
+
+function clean_up_hub_cluster_profile_artifacts {
+  
+  echo "************ telcov10n Get rid of those artifacts that would be used during Spoke deployments ************"
+
+  timeout -s 9 10m ssh "${SSHOPTS[@]}" "root@${AUX_HOST}" bash -s --  \
+  "${NAMESPACE}" "${SHARED_HUB_CLUSTER_PROFILE}" << 'EOF'
+set -o nounset
+set -o errexit
+set -o pipefail
+
+cluster_profile_shared_folder=/var/builds/${1}/${2}
+cluster_profile_symbolic_link=/var/telco-qe-preserved/${2}
+
+rm -fr ${cluster_profile_symbolic_link} ${cluster_profile_shared_folder}
+EOF
+}
+
+function pull_request_debug {
+
+  echo "Using pull request ${PULL_NUMBER}... DO NOT preserve the hub cluster"
+
+  echo "######################################################################"
+  echo "# From here WIP changes"
+  echo "######################################################################"
+  echo " To quit, run the following command from POD shell: "
+  echo " $ touch ${HOME}/debug.done"
+  echo "######################################################################"
+  echo
+
+  set -x
+  oc get no,clusterversion,mcp,co,sc,pv
+  oc get subscriptions.operators,OperatorGroup,pvc -A
+  oc whoami --show-console
+  set +x
+  echo "Current OCP Hub cluster namespace is ${NAMESPACE}"
+
+  set -x
+  while sleep 1m; do
+    date
+    test -f ${HOME}/debug.done && break
+  done
+
+}
+
+function main {
+  setup_aux_host_ssh_access
+  create_shared_hub_cluster_profile_artifacts_folder
+  save_hub_cluster_profile_artifacts
+}
+
+main
+
+if [ -n "${PULL_NUMBER:-}" ]; then
+  pull_request_debug
+  clean_up_hub_cluster_profile_artifacts
```

Comment:
@aleskandro to be able to debug, I am also copying the hub cluster profile into the shared location, but upon exiting, I would remove such folder to keep AUX_HOST disk as clean as possible.